awx/main/credential_plugins/hashivault.py

```diff
@@ -89,7 +89,6 @@ def kv_backend(**kwargs):
             params['version'] = kwargs['secret_version']
         try:
             mount_point, *path = pathlib.Path(secret_path.lstrip(os.sep)).parts
-            '/'.join(*path)
```

Comment:
@jainnikhil30 can you provide an exact reproducer for this bug that is fixed by this change?

I created a nested v2 secret using path `/kv/multivault/first` (key=foo) and, with this change, I'm getting a `404 - no handler for route` error.